back/src/test/java/com/ubikloadpack/jmeter/ulp/observability/log/SampleLogTest.java

```diff
@@ -0,0 +1,109 @@
+package com.ubikloadpack.jmeter.ulp.observability.log;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import java.util.Date;
+
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.ubikloadpack.jmeter.ulp.observability.log.SampleLog;
+import com.ubikloadpack.jmeter.ulp.observability.log.SampleLoggerTest;
+import com.ubikloadpack.jmeter.ulp.observability.metric.ResponseResult;
+import com.ubikloadpack.jmeter.ulp.observability.registry.MicrometerRegistry;
+import com.ubikloadpack.jmeter.ulp.observability.util.Util;
+
+import io.micrometer.core.instrument.distribution.ValueAtPercentile;
+
+public class SampleLogTest {
+	
+	@Test
+	public void testToOpenMetricsString() {
+        // Arrange
+        String sampleName = "testSample";
+        Date timeStamp = new Date();
+        Long samplerCount = 5L;
+        Long error = 1L;
+        ValueAtPercentile[] pct = new ValueAtPercentile[] { new ValueAtPercentile(0.9, 500), new ValueAtPercentile(0.95, 600) };
+        Long sum = 2500L;
+        Double avg = 500.0;
+        Long max = 600L;
+        Double throughput = 5.0;
+        Long threads = 1L;
+        Long samplerCountTotal = 5L;
+        Long maxTotal = 600L;
+        Double avgTotal = 500.0;
+        Long errorTotal = 1L;
+        Double throughputTotal = 5.0;
+        ValueAtPercentile[] pctTotal = new ValueAtPercentile[] { new ValueAtPercentile(0.9, 300), new ValueAtPercentile(0.95, 400) };
+        Long threadsTotal = 1L;
+
+        SampleLog sampleLog = new SampleLog(sampleName, timeStamp, samplerCount, error, pct, sum, avg, max, throughput, threads, 
+            samplerCountTotal, maxTotal, avgTotal, errorTotal, throughputTotal, pctTotal, threadsTotal);
+
+        String openMetrics = sampleLog.toOpenMetricsString();
+        
+        assertTrue(openMetrics != null);

```

Comment:
Always add a clear message explaining failure.
To see what you get, make the test fails and check if failure message is clear enough

Reply:
i did a mistake while adding this assertion, it doesn't make sense because the method toOpenMetricsString() never returns null. Instead i should check that the string is not empty.